### ISSUE TITLE
Decoding extrinsics from binary / hex

### ISSUE THREAD

**Issue by mutantcornholio** [polkadot-api/polkadot-api#684]
My usecase: the application using PAPI generates some extrinsic, which I can later pass somewhere in hex form, like this: https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frococo-rpc.polkadot.io#/extrinsics/decode/0x15002b0901a41203070010a5d4e800e8c3574bab0a67e9ae1e646fd60073bc629d2c518f78a5ece65bf55058abde42010a000000

In tests, I'd like to be able to take that `0x15002b0901a41203070010a5d4e800e8c3574bab0a67e9ae1e646fd60073bc629d2c518f78a5ece65bf55058abde42010a000000` string and decode it back, to validate the contents. Or at least, to tell that it's actually a valid extrinsic of expected pallet/method.

**Comment by carlosala** [comment]
Hey! We just released [this API](https://github.com/polkadot-api/polkadot-api/blob/4f9782e6e4e3f610df460dd64c3d3e77b46d50d0/packages/client/src/types.ts#L100) that allows to create transactions from a call data. Would that be enough for you?
If you want a more low level approach, you can use something similar to [this code](https://github.com/polkadot-api/polkadot-api/blob/4f9782e6e4e3f610df460dd64c3d3e77b46d50d0/packages/client/src/client.ts#L154-L161) that uses the dynamic builder.

**Comment by josepot** [comment]
> Hey! We just released [this API](https://github.com/polkadot-api/polkadot-api/blob/4f9782e6e4e3f610df460dd64c3d3e77b46d50d0/packages/client/src/types.ts#L100) that allows to create transactions from a call data.

This issue is about having the ability to decode extrinsics, which is something that I plan on adding into the [new tx-utils library](https://github.com/polkadot-api/polkadot-api/pull/673). The new API doesn't address this.

**Comment by carlosala** [comment]
Oh, _extrinsics_ not _call data_. Sorry for the confusion😆

**Comment by josepot** [comment]
> Actually, the example is not an extrinsic, it's just a _call data_.

You are right: it's unclear what Yuri is asking here, but we got soooo many requests asking us to have the ability to decode extrinsics that I want to have an issue to track it... and the title of the issue does describe that, so...

**Comment by carlosala** [comment]
Hahaha ok, let's take advantage of this issue for that😉👍

**Comment by mutantcornholio** [comment]
Yeah, I wasn't precise with vocablurary 😅
I only need call data for my case, so just doing this worked for me just fine:

```
const tx = await api.txFromCallData(Binary.fromHex(extrinsicHex!));
expect(tx.decodedCall.type).toEqual("Referenda");
expect(tx.decodedCall.value.type).toEqual("submit");
```

So, thanks ❤️

**Comment by carlosala** [comment]
Great! We'll keep this issue opened to track what Josep said 😉 

I'm glad we solved your particular use case 🎉 🚀
